src/components/StatePicker/index.js

```diff
@@ -45,15 +49,18 @@ function StatePicker({value, errorText, onInputChange, forwardedRef, label}) {
         setIsPickerVisible(true);
     };
 
-    const hidePickerModal = () => {
+    const hidePickerModal = (isUpdateStateInput = false) => {
+        if (!isUpdateStateInput) {
+            onBlur();
+        }
```

Comment:
- There are a few methods that can make the picker modal hidden, for example, clicking on back button, or clicking on any option.
- We only call the onBlur function if the modal is hidden when user clicks on the back button, because if the user did not open the picker modal before, then opened it, and clicked on any option, the modal will be hidden and the error message appears for a while